lib/vagrant/util/mime.rb

```diff
@@ -0,0 +1,69 @@
+require 'mime/types'
+require 'securerandom'
+
+module Vagrant
+  module Util
+    module Mime
+      class Multipart
+
+        attr_accessor :content
+
+        attr_accessor :content_type
+
+        attr_accessor :headers
+
+        attr_accessor :mime_version
+
+        def initialize(content_type="multipart/mixed", mime_version="1.0")
+          @content_id = "#{SecureRandom.alphanumeric(24)}@#{SecureRandom.alphanumeric(24)}.local"
```

Comment:
not too sure about this `content_id` bit
https://tools.ietf.org/html/rfc2392

Reply:
The format is fine since it adheres to the url-addr-spec. To ensure uniqueness it might not be a bad idea to swap out the first section with a `Time.now.to_i` so you end up with something like: 

```ruby
"#{Time.now.to_i}@#{SecureRandom.alphanumeric(24)}.local"
```